src/ctx.h

```diff
@@ -479,9 +479,12 @@ class FunctionEmitContext {
                                   const PointerType **resultPtrType = nullptr);
 
     /** Bool is stored as i8 and <WIDTH x i8> but represented in IR as i1 and
-     * <WIDTH x MASK>. This is a helper function to match bool size at storage
-     * interface. */
-    llvm::Value *SwitchBoolSize(llvm::Value *value, llvm::Type *toType, const llvm::Twine &name = "");
+     * <WIDTH x MASK>. This is a helper function to match bool size. The
+     * argument toStorageType is true when conversion to storage type is
+     * needed.  */
+    llvm::Value *SwitchBoolSize(llvm::Value *value, llvm::Type *toType, bool toStorageType,
```

Comment:
Personally I don't like bool function parameters because they reduce readability of the code quite a lot - you need go to function declaration to understand what this parameter does. I would do 2 separate functions `SwitchBoolSizeToI1` and `SwitchBoolSizeToStorageType`. 
But it's up to you. We already have so many functions with similar bool parameters so this change doesn't make any difference in this regard :)

Reply:
I have introduced two wrappers `SwitchBoolToStorageType` and `SwitchBoolToMaskType` that are to be used to convert types.